### PR TITLE
chore: remove compiled sources from wheels

### PR DESCRIPTION
**Pull request by P403n1x87**
We reduce the size of generated wheels by removing compiled souces, e.g. C/C++/Cython sources, which are not required for using the library. The measured size reduction is between 1 MB and 2 MB.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: .github/workflows/build_deploy.yml [hunk 1]
```diff
@@ -10,8 +10,14 @@ on:
       - build_deploy*
   pull_request:
     paths:
+      - .github/workflows/build_deploy.yml
```
Comment (P403n1x87) [inline .github/workflows/build_deploy.yml:13]: These changes should probably go in a separate PR

File: .github/workflows/build_python_3.yml [hunk 2]
```diff
@@ -55,6 +55,23 @@ jobs:
...
+        if : ${{ matrix.os == 'windows-latest' }}
+        run: |
+          $whl_files = Get-ChildItem -Path .\wheelhouse -Filter *.whl
+
+          foreach ($whl_file in $whl_files) {
+              Expand-Archive -Path $whl_file.FullName -DestinationPath .\temp -Force
+              Get-ChildItem -Path .\temp -Include *.c,*.cpp,*.cc,*.h,*.hpp,*.pyx -Recurse | Remove-Item -Force
+              Compress-Archive -Path .\temp\* -DestinationPath $whl_file.FullName -Force
+              Remove-Item -Path .\temp -Recurse -Force
+          }
```
Comment (P403n1x87) [inline .github/workflows/build_python_3.yml:73]: Suggested by Copilot 🤖

File: .github/workflows/build_deploy.yml (hunk 1 above)
Comment (P403n1x87) [reply .github/workflows/build_deploy.yml]: #7294

**Review by emmettbutler — DISMISSED** [review]

**Review by mabdinur — DISMISSED** [review]

**Review by majorgreys — APPROVED** [review]
